### ISSUE TITLE
Problem building nng for Raspberry Pi4

### ISSUE THREAD

**Issue by denis-troller** [nanomsg/nng#1029]
Note that by filing an issue, you agree that you have read and agreed to
our contribution guidelines.

## NNG & Platform details.

nng version: 1.1.1
platform: Raspberry Pi 4/Raspbian
Compiler: gcc

## Expected Behavior
When  building nng on such a platform, it should build without errors.

## Actual Behavior
Upon running the ninja command, build fails with several similar erros, the first of which is:

> [90/286] Linking C executable tests/compat_surveyttl
> FAILED: tests/compat_surveyttl 
> : && /usr/bin/cc -Wall -Wextra -fno-omit-frame-pointer  -rdynamic tests/CMakeFiles/compat_surveyttl.dir/compat_surveyttl.c.o tests/CMakeFiles/compat_surveyttl.dir/compat_testutil.c.o  -o tests/compat_surveyttl  libnng.a -lpthread && :
> /usr/bin/ld: libnng.a(posix_atomic.c.o): in function `nni_atomic_inc64':
> posix_atomic.c:(.text+0xbc): undefined reference to `__atomic_fetch_add_8'
> /usr/bin/ld: libnng.a(posix_atomic.c.o): in function `nni_atomic_dec64':
> posix_atomic.c:(.text+0xf4): undefined reference to `__atomic_fetch_sub_8'
> /usr/bin/ld: libnng.a(posix_atomic.c.o): in function `nni_atomic_get64':
> posix_atomic.c:(.text+0x124): undefined reference to `__atomic_load_8'
> /usr/bin/ld: libnng.a(posix_atomic.c.o): in function `nni_atomic_set64':
> posix_atomic.c:(.text+0x17c): undefined reference to `__atomic_store_8'
> /usr/bin/ld: libnng.a(posix_atomic.c.o): in function `nni_atomic_swap64':
> posix_atomic.c:(.text+0x1c0): undefined reference to `__atomic_exchange_8'
> /usr/bin/ld: libnng.a(posix_atomic.c.o): in function `nni_atomic_init64':
> posix_atomic.c:(.text+0x218): undefined reference to `__atomic_store_8'
> collect2: error: ld returned 1 exit status

## Steps to Reproduce
On a raspberry Pi 4 with Raspbian, obtain the source code and then
```
mkdir build
cd build
cmake -G Ninja ..
ninja
```

**Comment by jeikabu** [comment]
Seems to be in Debian Buster (which if I'm not mistaken Pi4 requires).
This is normally associated with clang, but maybe gcc 8 as well.  Anyway, you probably need `-latomic`.  You can probably throw the following into `src/CMakeLists.txt`:
```cmake
target_link_libraries (${PROJECT_NAME} PUBLIC atomic)
```

Guess a more general solution would involve `CheckAtomic.cmake`.

**Comment by gdamore** [comment]
Yeah, we need to add a check for this somewhere.  I've run into another situation where on some platforms symbols are in external libraries and others they are in libc.

**Comment by gdamore** [comment]
I believe this just got fixed by a contribution in PR #1196
